GA/HeaderFiles/Crossover.h

```diff
@@ -57,12 +59,12 @@ void Crossover(vector<vector<vector<float> > > & varInput, vector<vector<vector<
 					{
 						varOutput[i+reproduction_no][j][k] = varInput[locations[i]][j][k];
 						varOutput[i+1+reproduction_no][j][k] = varInput[locations[i+1]][j][k];
-	                }
+					}
 					else
-	                {
+					{
 						varOutput[i+reproduction_no][j][k] = varInput[locations[i+1]][j][k];
 						varOutput[i+1+reproduction_no][j][k] = varInput[locations[i]][j][k];
-	                }
+					}
```

Comment:
A couple of questions on these actually: 

1. Is there any particular reason spaces are preferred? I'm just not familiar with the difference so I would like to know before I implement.
2. Similarly, why might K&R be preferred? I have actually never seen this style personally so I would like to know what the advantages/disadvantages are.

Thanks for the feedback on this, I do want to make this as readable as possible.

Reply:
I will also ask the team in our working meeting tomorrow if they have any preferred method for this stuff.

Reply:
>     1. Is there any particular reason spaces are preferred? I'm just not familiar with the difference so I would like to know before I implement.

Mostly two reasons:
1. a space is a space everywhere, whereas a tab is often 8 characters wide when you look at it in a browser (e.g. here) but 4 characters on most editors.  On some editors it might still be 8 or some other values, meaning that the indentation might look different in different places.  Spaces always look the same.
2. Spaces allow some more flexibility, e.g. you can align things like:
    ```cpp
    func("first argument",
         "second argument")
   ```
    The second arg is aligned under the first using 5 spaces.  If you use tabs you wouldn't be able to do that (unless you mix tabs and spaces, which is even worse).

Keep in mind that you can configure your editor to automatically use spaces while indenting with the tab key on your keyboard, so you don't have to hit the spacebar (or del) 4 times.

>     2. Similarly, why might K&R be preferred? I have actually never seen this style personally so I would like to know what the advantages/disadvantages are.

It's just the most popular one.  Since the `{` is opened at the end of the current line, you don't "waste" a line for it, while the `}` is on a line on its own to mark the end of the block and separate it from the next.

> I will also ask the team in our working meeting tomorrow if they have any preferred method for this stuff.

Do you want me to join?  I might be able to attend (depending on what time it is.)

Reply:
Looks like Ryan and I prefer the Allman convention and always using brackets and 4 spaces for tab. Looks like we should use CamelCase for classes and then mixed case for functions and everything else. Constants should be all caps. We should look into using config files and linters like these: https://github.com/cpp-linter/cpp-linter https://github.com/cpp-linter/cpp-linter-hooks/blob/main/.pre-commit-hooks.yaml .

Reply:
> Looks like Ryan and I prefer the Allman convention and always using brackets and 4 spaces for tab. Looks like we should use CamelCase for classes and then mixed case for functions and everything else. Constants should be all caps. We should look into using config files and linters like these: https://github.com/cpp-linter/cpp-linter https://github.com/cpp-linter/cpp-linter-hooks/blob/main/.pre-commit-hooks.yaml .

I generally prefer K&R with PascalCase for classes, camelCase for functions, and snake_case for variables. However, I think snake_case might be more of a Python thing generally, so I would be fine with whatever.